src/FormInput.js

```diff
@@ -0,0 +1,71 @@
+import {useState} from "react";
+import open_eye from "./icons/open_eye.png"
+import close_eye from "./icons/close_eye.png"
+
+let password = null;
+const FormInput = ({label, type, required, name, validation}) => {
+  const [field, setField] = useState('');
+  const [fieldError, setFieldError] = useState(false);
+  const [fieldBlur, setFieldBlur] = useState(false);
+
+  const fieldHandler = (e) => {
+    if (e.target.name !== 'Confirm password' ? !validation.test(e.target.value) : (password !== e.target.value)) {
+      setFieldError(true);
+      e.target.style.borderColor = 'red';
+    } else {
+      setFieldError(false);
+      e.target.style.borderColor = '#0747a6';
+    }
+    setField(e.target.value);
+  }
+  
+  const blurHandler = (e) => {
+    if (e.target.name === label) {
+      setFieldBlur(true);
+    }
+    if (e.target.name === 'Password') {
+      password = e.target.value;
+    }
+  }
+
+  const showPassword = (e) => {
+    const prev = e.target.previousElementSibling;
+    if (prev.type === 'password') {
+      prev.type = 'text';
+      e.target.src = open_eye;
+    } else {
+      prev.type = 'password';
+      e.target.src = close_eye;
+    }
+  }
+
+  return (
+    <div style={{margin: '20px 0'}}>
+      <label style={{display: 'block', fontWeight: 'bold', margin: '5px 0'}}>{label}</label>
+      <input onChange={e => fieldHandler(e)}
+             onBlur={e => blurHandler(e)}
+             name={label}
+             value={field}
+             type={type}
```

Comment:
`type={ isOpen ? "text" : "password" }`

where `isOpen` - value from `useState`-hook, and settled at the `showPassword`-function